diff --git a/gps.h b/gps.h
index 612b6ca8..a185917d 100644
--- a/gps.h
+++ b/gps.h
@@ -24,6 +24,7 @@ extern "C" {
 
 
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/time.h>
 #include <stdbool.h>
 #include <inttypes.h>	/* stdint.h would be smaller but not all have it */
diff --git a/libgps_core.c b/libgps_core.c
index d0a65f56..82c7a386 100644
--- a/libgps_core.c
+++ b/libgps_core.c
@@ -16,6 +16,7 @@
 #include <assert.h>
 #include <sys/time.h>	 /* expected to have a select(2) prototype a la SuS */
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #ifndef S_SPLINT_S
 #include <sys/socket.h>
